src/analysis/special_functions/polynomials.lean

```diff
@@ -0,0 +1,174 @@
+/-
+Copyright (c) 2020 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+import analysis.asymptotic_equivalent
+import data.polynomial.erase_lead
+
+/-!
+# Limits related to polynomial and rational functions
```

Comment:
I started to leave line comments below, but then I felt there is something sub-optimal in the large scale structure of this file.

First I think there are missing supporting lemmas for equivalence (in other files). Probably including:
```lean

section

variables {α β : Type*} [normed_group β]

lemma asymptotics.is_equivalent.congr_left {u v w : α → β} {l : filter α} (huv : u ~[l] v) (huw : u =ᶠ[l] w) : w ~[l] v :=
is_o.congr' (huw.sub (eventually_eq.refl _ _)) (eventually_eq.refl _ _) huv

lemma asymptotics.is_equivalent.congr_right {u v w : α → β} {l : filter α} (huv : u ~[l] v) (hvw : v =ᶠ[l] w) : u ~[l] w :=
(huv.symm.congr_left hvw).symm

lemma filter.eventually_eq.sub_eq {u v : α → β} {l : filter α} (h : u =ᶠ[l] v) : u - v =ᶠ[l] 0 :=
by simpa using (eventually_eq.sub (eventually_eq.refl l u) h).symm

lemma filter.eventually_eq_iff_sub {u v : α → β} {l : filter α} : u =ᶠ[l] v ↔ u - v =ᶠ[l] 0 :=
⟨λ h, h.sub_eq, λ h, by simpa using h.add (eventually_eq.refl l v)⟩

lemma eventually_eq.is_equivalent {u v : α → β} {l : filter α} (h : u =ᶠ[l] v) : u ~[l] v :=
is_o.congr' h.sub_eq.symm (eventually_eq.refl _ _) (is_o_zero v l)
end 
```
Then I think the first lemma about quotients of polynomial should be:
```lean
lemma eval_div_equivalent_at_top [normed_linear_ordered_field α]
  [order_topology α] (P Q : polynomial α) :
   (λ x, (eval x P)/(eval x Q)) ~[at_top] λ x, P.leading_coeff/Q.leading_coeff * x^(P.nat_degree - Q.nat_degree : ℤ) :=
begin
  by_cases hP : P = 0,
  { simp [hP] },
  by_cases hQ : Q = 0,
  { simp [hQ] },
  refine (P.eval_is_equivalent_at_top_eval_lead.symm.div 
          Q.eval_is_equivalent_at_top_eval_lead.symm).symm.trans
         (eventually_eq.is_equivalent ((eventually_gt_at_top 0).mono $ λ x hx, _)),
  simp [← div_mul_div, hP, hQ, fpow_sub (ne_of_gt hx)],
end
```
Note how this lemma makes no assumption at all about `P` and `Q`, and uses `fpow` (allowing for negative powers).